pygmt/base_plotting.py

```diff
@@ -855,25 +855,50 @@ def text(
             the text
         text : str or 1d array
             The text string, or an array of strings to plot on the figure
-        angle: int, float or bool
+        angle: int, float, str or bool
             Set the angle measured in degrees counter-clockwise from
             horizontal. E.g. 30 sets the text at 30 degrees. If no angle is
-            given then the input textfile(s) must have this as a column.
+            explicitly given (i.e. angle=True) then the input textfile(s) must
+            have this as a column.
         font : str or bool
             Set the font specification with format "size,font,color" where size
             is text size in points, font is the font to use, and color sets the
             font color. E.g. "12p,Helvetica-Bold,red" selects a 12p red
-            Helvetica-Bold font. If no font info is given then the input
-            textfile(s) must have this information in one of its columns.
-        justify: str or bool
+            Helvetica-Bold font. If no font info is explicitly given (i.e.
+            font=True), then the input textfile(s) must have this information
+            in one of its columns.
+        justify : str or bool
             Set the alignment which refers to the part of the text string that
             will be mapped onto the (x,y) point. Choose a 2 character
             combination of L, C, R (for left, center, or right) and T, M, B for
             top, middle, or bottom. E.g., BL for lower left. If no
-            justification is given then the input textfile(s) must have this as
-            a column.
+            justification is explicitly given (i.e. justify=True), then the
+            input textfile(s) must have this as a column.
         {J}
         {R}
+        clearance : str
+            ``[dx/dy][+to|O|c|C]``
+            Adjust the clearance between the text and the surrounding box
+            [15%]. Only used if *pen* or *fill* are specified. Append the unit
+            you want ('c' for cm, 'i' for inch, or 'p' for point; if not given
+            we consult 'PROJ_LENGTH_UNIT') or '%' for a percentage of the
+            font size. Optionally, use modifier '+t' to set the shape of the
+            textbox when using *fill* and/or *pen*. Append lower case 'o' to
+            get a straight rectangle [Default]. Append upper case 'O' to get a
+            rounded rectangle. In paragraph mode (*paragraph*) you can also
+            append lower case 'c' to get a concave rectangle or append upper
+            case 'C' to get a convex rectangle.
+        fill : str
+            Sets the shade or color used for filling the text box [Default is
+            no fill]. Alternatively, use fill='c' to plot the text and then use
+            the text dimensions (and *clearance*) to build clip paths and turn
+            clipping on. This clipping can then be turned off later with clip
+            -C. To not plot the text but activate clipping, use fill='C'
```

Comment:
```suggestion
            no fill].
```

Probably should remove this section? Since [`clip`](https://docs.generic-mapping-tools.org/latest/clip.html) isn't implemented yet, and I'm not even sure if I understand what it does. Or we can just keep it in case someone does need to use it.

Reply:
The text clipping feature won't work without the **clip** function. I think it's OK to remove this part.

Reply:
Ok done. Technically users can still use `fill='c'` as we're just not documenting it. But best not to confuse people what the `clip` thing is about until it is implemented.